dbt/adapters/spark_livy/connections.py

```diff
@@ -592,6 +597,40 @@ def close(cls, connection):
         except Exception as err:
             logger.debug(f"Error closing connection {err}")
 
+    @classmethod
+    def fetch_spark_version(cls, connection):
+
+        if SparkConnectionManager.spark_version: 
+            return SparkConnectionManager.spark_version
+
+        try:
+            sql = "select version()"
+            cursor = connection.handle.cursor()
+            cursor.execute(sql)
+
+            res = cursor.fetchall()
+
+            SparkConnectionManager.spark_version = res[0][0].split(".")[0]
+
+            payload = {
+                "event_type": "dbt_spark_livy_warehouse",
+                "warehouse_version": { "version": SparkConnectionManager.spark_version, "build": res[0][0] },
+            }
+            tracker.track_usage(payload)
+        except Exception as ex:
+            # we couldn't get the spark warehouse version, default to version 2
+            logger.debug(f"Cannot get spark version, defaulting to version 2. Error: {ex}")
+            SparkConnectionManager.spark_version = "2"
+
+            payload = {
+                "event_type": "dbt_spark_livy_warehouse",
+                "warehouse_version": { "version": "2", "build": "NA" },
+            }
+            tracker.track_usage(payload)
+
+        os.environ['DBT_SPARK_VERSION'] = SparkConnectionManager.spark_version 
```

Comment:
should we set a default for DBR_SPARK_VERSION as 2 or 3.

Reply:
reason I have set this as 2 is because the version() call fails in 2